lib/modules/showImages.js

```diff
@@ -638,292 +569,244 @@ function domainToModuleName(hostname) {
+	// and instead expand ours
+	if (thing.querySelector('.media-preview')) {
+		thing.querySelector('.media-preview').remove();
+		expando.expandWanted = true;
+	}
 
-		if (expandoButton.mediaElement.collapse) {
-			const removeInstead = expandoButton.mediaElement.collapse();
-			if (removeInstead) {
-				expandoButton.destroy();
-			}
-		}
+	if (thing.isComment()) {
+		let wasOpen;
+
+		// Execute expando toggle procedure when comment collapse / expand
+		thing.$thing
+			.parents('.comment')
+			.andSelf()
+			.find('> .entry .expand')
+			.click(() => {
+				if (expando.box.offsetParent) {
+					if (wasOpen) expando.expand();
+				} else {
+					wasOpen = expando.open;
+					if (expando.open) expando.collapse();
+				}
+
+				updateAutoExpandCount();
+			});
 	}
 
-	if (isPageType('comments')) {
-		// Execute expando collapse procedure when comment is collapsed
-		$(expandoButton).parents('.comment').find('> .entry .tagline > .expand').click(collapse);
+	if (module.options.autoMaxHeight.value && !expando.isEntryLink) {
+		thing.entry.addEventListener('mediaResize', updateParentHeight);
 	}
 
-	if (showHide) expand();
-	else collapse();
-}
+	expando.resize = frameDebounce(() => {
+		const $container = $(expando.box).closest('.md > *, .entry > *');
+		// jQuery().width() returns non-padded width
+		const width = $container.parent().width();
+		// In case the box is inside a table or list, normalize its horizontal position
+		const marginLeft = $container[0].getBoundingClientRect().left -
+			expando.box.getBoundingClientRect().left;
 
-async function convertGifToVideo(options) {
-	const { gfyName: id } = await ajax({
-		type: 'json',
-		url: '//upload.gfycat.com/transcodeRelease',
-		data: { fetchUrl: options.src },
-		cacheFor: DAY,
+		expando.adjustInnersize(width, marginLeft);
 	});
 
-	if (id) {
-		return await gfycat.handleLink('', [undefined, id]);
-	} else {
-		throw Error('Could not convert gif to video', options.src);
+	const withinThing =
+		module.options.autoExpandSelfText.value && thing.isSelfPost() && !this.isEntryLink;
+
+	if (!expando.expandWanted) expando.expandWanted = expando::isExpandWanted(withinThing ? thing : null);
+	const primary = expando.getPrimary();
+	// Do not expand if duplicate and primary is open
+	if (primary && primary.isAttached() && (primary.open || primary.expandWanted)) {
+		expando.expandWanted = false;
 	}
-}
 
-function buildExpandoBox(expandoButton) {
-	const imageLink = expandoButton.imageLink;
+	expando.initialize(options);
 
-	const options = {
-		loadHeight: 20, // gives an indication that the expando is opened
-	};
+	updateAutoExpandCount();
+}
 
-	const expando = expandoButton.expandoBox = $(expandoTemplate(options))[0];
-	const mediaContainer = expando.querySelector('.res-expando-media');
+function retrieveExpandoOptions(thing, { siteModule, detectResult, mediaLink, href }) {
+	return new Promise(async (resolve, reject) => {
```

Comment:
This promise is probably not necessary if you make `retrieveExpandoOptions` async...